reflect.Value.Set using unaddressable value

The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
)

func main() {
	r := uint32(2000000000)
	r = hello(r)

	fmt.Println(r)
}

func hello(r uint32) uint32 {
	return r + 1
}

```

Expected result:
```console
$ go run ./sample.go
2000000001
```

Got:
```console
$ yaegi ./sample.go
panic: reflect: reflect.Value.Set using unaddressable value [recovered]
        panic: reflect: reflect.Value.Set using unaddressable value

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc0000c7770)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/run.go:105 +0x134
panic(0xbaab80, 0xc000152770)
        /home/ldez/.gvm/gos/go1.12.4/src/runtime/panic.go:522 +0x1b5
reflect.flag.mustBeAssignable(0x8a)
        /home/ldez/.gvm/gos/go1.12.4/src/reflect/value.go:234 +0x13a
reflect.Value.Set(0xbaae00, 0xc0000bc7c8, 0x8a, 0xbaae00, 0xc0000bc7d4, 0x18a)
        /home/ldez/.gvm/gos/go1.12.4/src/reflect/value.go:1467 +0x2f
github.com/containous/yaegi/interp.call.func4(0xc0000c7770, 0xc0000d9c70)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/run.go:604 +0x946
github.com/containous/yaegi/interp.runCfg(0xc0000f6d20, 0xc0000c7770)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/run.go:110 +0x62
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0000c2660, 0xc0000f65a0, 0xc0000c7590)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/run.go:92 +0x126
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000c2660, 0xc0001b4000, 0x170, 0x370, 0xc0001b4000, 0x170, 0x0, 0xc000194340)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:269 +0x43c
main.main()
        /home/ldez/sources/go/src/github.com/containous/yaegi/yaegi.go:59 +0x3ec
```

--- a/_test/assign2.go
+++ b/_test/assign2.go
@@ -1,0 +1,12 @@
+package main
+
+func main() {
+	r := uint32(2000000000)
+	r = hello(r)
+	println(r)
+}
+
+func hello(r uint32) uint32 { return r + 1 }
+
+// Output:
+// 2000000001

--- a/interp/run.go
+++ b/interp/run.go
@@ -170,14 +170,14 @@ func typeAssert2(n *Node) {
 }
 
 func convert(n *Node) {
-	i := n.findex
+	dest := genValue(n)
 	c := n.child[1]
 	typ := n.child[0].typ.TypeOf()
 	next := getExec(n.tnext)
 
 	if c.kind == BasicLit && c.val == nil { // convert nil to type
 		n.exec = func(f *Frame) Builtin {
-			f.data[i] = reflect.New(typ).Elem()
+			dest(f).Set(reflect.New(typ).Elem())
 			return next
 		}
 		return
@@ -191,7 +191,7 @@ func convert(n *Node) {
 	}
 
 	n.exec = func(f *Frame) Builtin {
-		f.data[i] = value(f).Convert(typ)
+		dest(f).Set(value(f).Convert(typ))
 		return next
 	}
 }

